packages/app-builder/src/models/ast-node.ts

```diff
@@ -524,22 +530,60 @@ export function isOrWithAndAstNode(
   return true;
 }
 
-export interface TwoLineOperandAstNode {
-  name: TwoLineOperandOperatorFunction;
+export interface MainAstNode {
+  name: MainAstOperatorFunction;
+  constant: undefined;
+  children: AstNode[];
+  namedChildren: Record<string, never>;
+}
+
+export interface MainAstBinaryNode {
+  name: BinaryMainAstOperatorFunction;
   constant: undefined;
   children: [AstNode, AstNode];
   namedChildren: Record<string, never>;
 }
 
-export function isTwoLineOperandAstNode(
-  astNode: AstNode,
-): astNode is TwoLineOperandAstNode {
-  if (isLeafOperandAstNode(astNode)) return false;
+export interface MainAstUnaryNode {
+  name: UnaryMainAstOperatorFunction;
+  constant: undefined;
+  children: [AstNode];
+  namedChildren: Record<string, never>;
+}
 
-  if (!hasExactlyTwoElements(astNode.children)) return false;
-  if (Object.keys(astNode.namedChildren).length > 0) return false;
-  if (astNode.name == null || !isTwoLineOperandOperatorFunction(astNode.name))
+export function isMainAstNode(astNode: AstNode): astNode is MainAstNode {
+  if (isLeafOperandAstNode(astNode)) {
+    return false;
+  }
+
+  if (Object.keys(astNode.namedChildren).length > 0) {
     return false;
+  }
+  if (astNode.name == null || !isMainAstOperatorFunction(astNode.name)) {
+    return false;
+  }
 
   return true;
 }
+
+export function isMainAstUnaryNode(
+  astNode: AstNode,
+): astNode is MainAstUnaryNode {
```

Comment:
Typescript question here.
This indication on the return value is never _enforced_ anywhere, right ? It's just something that tells whoever _consumes_ its output that it has this expected form.
Which means that, if the implementation of the function runs against the declared output type, I'm liable to get unexpected prod errors, correct ?

Reply:
Yes.
Typescript blindly accept the type predicate and use it in static type analysys.
The only help you have from TS here is the certainty the implementation is valid TS code, but you can write wrong type predicates
```ts
function isNumber(value: string | number): value is number {
    return typeof value === 'string';
}

// example of help you can get from TS, which is not related to the predicate
function isNumber(value: string | number): value is number {
  return value.prop === 'number'; // Property 'prop' does not exist on type 'string | number'
}

```

Reply:
Yes if the type narrowing goes wrong you could get prod errors on this one.